velox/exec/Spill.cpp

```diff
@@ -133,7 +133,12 @@ void SpillState::setPartitionSpilled(uint32_t partition) {
 void SpillState::validateSpillBytesSize(uint64_t bytes) {
   static constexpr uint64_t kMaxSpillBytesPerWrite =
       std::numeric_limits<int32_t>::max();
-  VELOX_CHECK_LT(bytes, kMaxSpillBytesPerWrite, "Spill bytes will overflow.");
+  if (bytes >= kMaxSpillBytesPerWrite) {
+    VELOX_SPILL_LIMIT_EXCEEDED(fmt::format(
```

Comment:
This is a separate limit than max spill limit. This is more like to handle the size limitation enforced by presto serializer but not spill itself.